src/customJQuery.js

```diff
@@ -0,0 +1,159 @@
+    innerObject.element = document.querySelectorAll(elem);
+    innerObject.array = [];
+
+    innerObject.toString = function() {
+        return innerObject.array.toString();
+    }
+
+    innerObject.addClass = function(...classNames) {
+        
+        innerObject.element.forEach(obj => {
+
+            classNames.forEach( className => {
+                obj.classList.add(className);
+            });
+
+        })
+    };
+
+    innerObject.removeClass = function(...classNames) {
+        
+        innerObject.element.forEach(obj => {
+            classNames.forEach( className => {
+                obj.classList.remove(className);
+            });
+        });
+    };
+
+    innerObject.append = function(...contentData) {
+
+        innerObject.element.forEach(obj => {
+            contentData.forEach(contentInfo => {
+                const tempInnerHtml = obj.innerHTML;
+                obj.innerHTML = tempInnerHtml + contentInfo;
+            });
+        });
+    };
+
+    innerObject.remove = function(selector) {
+
+        innerObject.element.forEach(obj => {
+            if(Object.prototype.toString.call(selector) === '[object Undefined]'){
+                obj.remove();
+            } else {
+                obj.remove(selector);
+            }
+        })
+    };
+
+    innerObject.text = function(str) {
+
+        if (str === undefined){
+            let res = '';
+            innerObject.element.forEach(obj => {
+               res += obj.textContent + '\n';
+            })
+            return res;
+        }
+        innerObject.element.forEach(obj => {
+            obj.textContent = str;
+        })
+        
+    };
+
+    innerObject.attr = function(attributeName, value){
+
+        if(value === undefined){
+
+            if(Object.prototype.toString.call(attributeName) == '[object Object]'){
+                for (let key in attributeName){
+                    innerObject.element.forEach(obj => {
+                        obj.setAttribute(key, attributeName[key]);
+                    })
+                }
+            } else if(innerObject.element[0].hasAttribute(attributeName)){
+                return innerObject.element[0].getAttribute(attributeName);
+            } 
+        } else {
+            innerObject.element.forEach(
+                obj => {
+
+                    if(Object.prototype.toString.call(value) == '[object Function]'){
+                        value = value();
+                    } 
+                    obj.setAttribute(attributeName, value);
+                }
+            );
+        }
+    };
+
+    innerObject.children = function() {
+        innerObject.array = [];
+        innerObject.element.forEach( obj => {
+            for ( let child of obj.children ) {
+                innerObject.array.push(child.nodeName);
+            }
+
+        });
+
+        return innerObject;
+    };
+
+    innerObject.empty = function() {
+        innerObject.element.forEach( obj => obj.innerHTML = '' );
```

Comment:
Good choice. Just a question. Do you know other possible ways to implement that? And why this way is better?